### ISSUE TITLE
Misleading error message when missing a colon

### ISSUE THREAD

**Issue by zmitchell** [rust-lang/rust#113873]
### Code

```Rust
use std::env;

#[derive(Debug, thiserror::Error)]
pub enum UriParseError {
    #[error("failed to locate 'parser_util' binary, is PARSER_UTIL_BIN set?")]
    BinNotFound(#[from] env::VarError),
}

/// Reads the `PARSER_BIN_UTIL` environment variable to get the path to the `parser_util` binary.
pub fn get_bin() -> Result<String, UriParseError> {
    env:var("PARSER_UTIL_BIN")
}
```

### Current output

```Shell
error: struct literal body without path
  --> crates/runix/src/uri_parser.rs:10:51
   |
10 |   pub fn get_bin() -> Result<String, UriParseError> {
   |  ___________________________________________________^
11 | |     env:var("PARSER_UTIL_BIN")
12 | | }
   | |_^
   |
help: you might have forgotten to add the struct literal inside the block
   |
10 ~ pub fn get_bin() -> Result<String, UriParseError> { SomeStruct {
11 |     env:var("PARSER_UTIL_BIN")
12 ~ } }
   |
```

### Desired output

```Shell
error: missing colon in path separator
  --> crates/runix/src/uri_parser.rs:10:51
   |
10 |   pub fn get_bin() -> Result<String, UriParseError> {
11 | |     env:var("PARSER_UTIL_BIN")
   |       ___^
12 | | }
   | |_^
   |
help: you might have forgotten to include a second colon
   |
10   pub fn get_bin() -> Result<String, UriParseError> {
11 ~     env::var("PARSER_UTIL_BIN")
12   }
   |
```
